types/react-mentions/index.d.ts

```diff
@@ -104,7 +105,12 @@ export type DisplayTransformFunc = (id: string, display: string, type: string) =
 /**
  * Defines the function signature for implementing @see MentionsInputProps.onChange
  */
-export type OnChangeHandlerFunc = (event: { target: { value: string } }, newValue: string, newPlainTextValue: string, mentions: MentionItem[]) => void;
+export type OnChangeHandlerFunc = (
+    event: React.ChangeEvent<HTMLTextAreaElement> | React.ChangeEvent<HTMLInputElement>,
```

Comment:
@efedorenko First, thanks for another contrib on these types! I hate to be a naysayer, but I'm not sure about this change due to the following line in react-mentions: https://github.com/signavio/react-mentions/blob/c7043f57cc5dddec2b3b4d03baa5d625b0731a57/src/MentionsInput.js#L363 Note there that the underlying code doesn't actually pass a real event & element, but an `eventMock` where the only actual property on the underlying target object is in fact `value` of type string. So while somewhat unintuitive, I think the type definition here is a faithful representation of the actual `react-mentions` code.

Also I mocked up a quick test for this using the old onChange event type to test against the type in this PR and the test seems to indicate that this change is incompatible.

Let me know if I'm missing something and I can take a closer look tomorrow.